openhcl/underhill_core/src/worker.rs

```diff
@@ -2831,6 +2859,81 @@ async fn new_underhill_vm(
                     .context("failed to connect to vmbus")?
             };
 
+            let mut filter = vmbus_client::filter::ClientFilterBuilder::new();
+
+            let mut vpci_filter = vmbus_client::filter::FilterDefinition::new("vpci")
+                .by_interface(guid::guid!("44C4F61D-4444-4400-9D52-802E27EDE19F"));
+
+            if enable_vpci_relay {
+                filter.add(&mut vpci_filter);
+            }
+
+            let mut relay_filter = vmbus_client::filter::FilterDefinition::new("relay").rest();
+            filter.add(&mut relay_filter);
+            vmbus_filter = Some(filter.build(tp, connection));
+
+            let connection = relay_filter.take();
+
+            if enable_vpci_relay {
+                use vpci_relay::*;
+
+                let mut relay = VpciRelay::new(
+                    driver_source.clone(),
+                    vpci_filter.take(),
+                    vmbus.control().clone(),
+                    dma_manager.new_client(DmaClientParameters {
+                        device_name: "vpci-relay".into(),
+                        lower_vtl_policy: LowerVtlPermissionPolicy::Vtl0,
+                        allocation_visibility: if hardware_isolated {
+                            AllocationVisibility::Shared
+                        } else {
+                            AllocationVisibility::Private
+                        },
+                        persistent_allocations: false,
+                    })?,
+                    vpci_relay_mmio,
+                    if use_mmio_hypercalls {
+                        Box::new(
+                            linux_mmio::HypercallMmio::new()
+                                .context("failed to create hypercall mmio accessor")?,
+                        )
+                    } else {
+                        Box::new(
+                            linux_mmio::DirectMmio::new()
+                                .context("failed to create direct mmio accessor")?,
+                        )
+                    },
+                );
+
+                // Allow NVMe devices.
+                relay.add_allowed_device(AllowedDevice {
+                    vendor_id: None,
+                    device_id: None,
+                    revision_id: None,
+                    prog_if: Some(
+                        ProgrammingInterface::MASS_STORAGE_CONTROLLER_NON_VOLATILE_MEMORY_NVME,
+                    ),
+                    sub_class: Some(Subclass::MASS_STORAGE_CONTROLLER_NON_VOLATILE_MEMORY),
+                    base_class: Some(ClassCode::MASS_STORAGE_CONTROLLER),
+                    sub_vendor_id: None,
+                    sub_system_id: None,
+                });
+
+                // Allow MANA devices.
+                relay.add_allowed_device(AllowedDevice {
+                    vendor_id: Some(0x1414),
```

Comment:
nit: can we lift these into some named constant even just the line above so it's more obvious (i think MSFT vendor, then MANA device_id?)

Reply:
OK